### ISSUE TITLE
`NetworkSubnet`: Attribute must be a whole number

### ISSUE THREAD

**Issue by yellowhat** [pulumi/pulumi-hcloud#505]
### What happened?

Hi,
consider the following example code.

With the latest `1.19.0`, on `pulumi up` I get the following error:

```console
Diagnostics:
  hcloud:index:NetworkSubnet (foonet):
    error: hcloud:index/networkSubnet:NetworkSubnet resource 'foonet' has a problem: Attribute must be a whole number, got 4229314. Examine values at 'foonet.networkId'.
```

Instead if I downgrade to `1.18.1` it works as expected.

Thanks

### Example

```python
import pulumi_hcloud as hcloud

mynet = hcloud.Network(
    "mynet",
    name="my-net",
    ip_range="10.0.0.0/8",
)

foonet = hcloud.NetworkSubnet(
    "foonet",
    network_id=mynet.id,
    type="cloud",
    network_zone="eu-central",
    ip_range="10.0.1.0/24",
)
```

### Output of `pulumi about`

```
CLI          
Version      3.115.2
Go Version   go1.22.2
Go Compiler  gc

Plugins
KIND      NAME        VERSION
resource  command     0.10.0
resource  hcloud      1.19.0
resource  kubernetes  4.11.0
language  python      unknown

Host     
OS       fedora
Version  40
Arch     x86_64

This project is written in python: executable='/usr/bin/python3' version='3.12.3'

Current Stack: organization/hetzner-k8s/dev

TYPE                          URN
pulumi:pulumi:Stack           urn:pulumi:dev::hetzner-k8s::pulumi:pulumi:Stack::hetzner-k8s-dev
pulumi:providers:hcloud       urn:pulumi:dev::hetzner-k8s::pulumi:providers:hcloud::default_1_19_0
hcloud:index/network:Network  urn:pulumi:dev::hetzner-k8s::hcloud:index/network:Network::mynet


Found no pending operations associated with dev

Backend        
Name           1092c12381af
URL            file://~
User           root
Organizations  
Token type     personal

Dependencies:
NAME               VERSION
ansible-core       2.16.6
argcomplete        3.3.0
kubernetes         29.0.0
libdnf5            5.1.17
pip                23.3.2
pulumi_command     0.10.0
pulumi_hcloud      1.19.0
pulumi_kubernetes  4.11.0
wheel              0.41.2

Pulumi locates its logs in /tmp by default
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by VenelinMartinov** [comment]
Hi @yellowhat, sorry you are having issues here and thanks for reporting! This looks like https://github.com/pulumi/pulumi-terraform-bridge/issues/1940 which is fixed on main but has not yet been released.

I've just kicked off a release and this will be fixed in 1.19.1 after https://github.com/pulumi/pulumi-hcloud/actions/runs/9062334519 completes

**Comment by VenelinMartinov** [comment]
https://github.com/pulumi/pulumi-hcloud/releases/tag/v1.19.1 was released which includes a fix for the problem. LMK if you are still having issues with this.
